app/controllers/teacher_interface/application_forms_controller.rb

```diff
@@ -34,7 +34,14 @@ def edit
     end
 
     def update
-      @application_form.submit! if @application_form.can_submit?
+      if application_form.can_submit?
+        application_form.submit!
```

Comment:
Happy with this as is for now but what do you think about moving this sort of stuff into services. I'm not sure it's a massive benefit one way or the other here but just to establish a pattern of encapsulating this sort of 'multiple things going on' operation.

Reply:
I think it probably is worth doing, I'll amend this PR to include it.